crypto/evp/evp_rand.c

```diff
@@ -0,0 +1,556 @@
+/*
+ * Copyright 2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/evp.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include "internal/cryptlib.h"
+#include <openssl/engine.h>
+#include <openssl/evp.h>
+#include <openssl/x509v3.h>
+#include <openssl/rand.h>
+#include <openssl/core.h>
+#include <openssl/core_names.h>
+#include <openssl/crypto.h>
+#include "crypto/asn1.h"
+#include "crypto/evp.h"
+#include "internal/numbers.h"
+#include "internal/provider.h"
+#include "evp_local.h"
+
+static int evp_rand_up_ref(void *vrand)
+{
+    EVP_RAND *rand = (EVP_RAND *)vrand;
+    int ref = 0;
+
+    if (rand != NULL)
+        return CRYPTO_UP_REF(&rand->refcnt, &ref, rand->lock);
+    return 1;
```

Comment:
This should be `return 0`

Reply:
Although, frankly, considering this is an internal function, if *our code* calls it with NULL, it's an implementation error.  Therefore, I wouldn't even check for NULL here.

Reply:
I'm happy with returning success for a NULL argument.  This is wrapped directly by EVP_RAND_up_ref().

Reply:
> This is wrapped directly by EVP_RAND_up_ref().

Considering that `EVP_RAND_up_ref()` does nothing extra, that's neither here nor there.

I just did a grep of all the `_up_ref` functions to have a look, and *none* of them return success on a NULL argument.  What makes this one different?  Or are you proposing that we should, in fact, return success on a NULL argument on all of them?  Why?

> I'm happy with returning success for a NULL argument.

I'm not.  It's quite a breach of current patterns, and as such, I think it requires better justification than "I'm happy"

Do note that I've nothing against such breaches, but I want to know the reasoning / justification, and if we can, it would be nice if such a change can be made across the board (I assume a separate PR).  I do have something against breaches that has potential of creating confusion ("wait, *that* `_up_ref` returns failure on NULL argument, but this one returns success?  What the hell???")